### PR TITLE
Change variables named `value` to other variable names

### PR DESCRIPTION
**Pull request by mcallic2**
**Description**
Fortran added the `value` type declaration in 2003. This PR suggests changes to these variable names, and does not include any directories/modules that are either no longer used and/or deprecated (i.e. axis_utils, mpp_io, fms_io, etc). 

Fixes ##1328

**How Has This Been Tested?**
autotools with gcc/13.1.0, netcdf/4.9.2, mpich/4.1.2, hdf5/1.14.1-2, libyaml/0.2.5

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

**Review by bensonr — COMMENTED** [review]
In axis_utils2.inc, you are keeping the 'value' name, where as the others are shortened to _x_val where the _x_ is shorthand for the type.  I think it would be good to stick to a single convention throughout.

**Comment by mcallic2** [comment]
@bensonr I love me some cohesiveness, these variable names are changed i n d7e68fd7

File: axis_utils/include/axis_utils2.inc [hunk 1]
```diff
@@ -242,17 +242,17 @@
        endif
     enddo
 
-    if (value < array(1) .or. value > array(ia)) then
- !       if (value < array(1))  frac_index = 1.
- !       if (value > array(ia)) frac_index = float(ia)
+    if (rval < array(1) .or. rval > array(ia)) then
+ !       if (frac_value < array(1))  frac_index = 1.
+ !       if (frac_value > array(ia)) frac_index = float(ia)
```
Comment (bensonr) [inline axis_utils/include/axis_utils2.inc:247]: Do you need to change these too?

Comment (mcallic2) [reply axis_utils/include/axis_utils2.inc]: those are comments so I must have not noticed them, good catch. updated in 2e22913c

Comment (mcallic2) [reply axis_utils/include/axis_utils2.inc]: yes sir, removed in 77e6d840

**Review by bensonr — APPROVED** [review]

**Review by rem1776 — APPROVED** [review]

**Review by rem1776 — APPROVED** [review]

File: mosaic2/mosaic2.F90 [hunk 2]
```diff
@@ -375,34 +375,34 @@ function transfer_to_model_index(istart, iend, refine_ratio)
...
+       parse_string = parse_string + 1
+       if(parse_string>nelem) then
          call mpp_error(FATAL, "mosaic_mod(parse_string) : number of element is greater than size(value(:))")
-      endif
-      last = first - 1 + scan(string(first:length), set)
-      if(last == first-1 ) then  ! not found, end of string
-         value(parse_string) = string(first:length)
+       endif
+       last = first - 1 + scan(string(first:length), set)
+       if(last == first-1 ) then  ! not found, end of string
```
Comment (bensonr) [inline mosaic2/mosaic2.F90:398]: Spacing changes here that need to be reverted for uniformity within the file.

Comment (mcallic2) [reply mosaic2/mosaic2.F90]: should be fixed in 02b33c617615a102c569504e8e0ef3bc5e53f22d

Comment (rem1776) [reply mosaic2/mosaic2.F90]: @mcallic2 There seems to still be a conflict with main, could you merge it in?

**Review by bensonr — DISMISSED** [review]

**Comment by rem1776** [comment]
@mcallic2 looks like an old `value` snuck in from the main merge:
```
mosaic2.F90(399): error #6303: The assignment operation or the binary expression operation is invalid for the data types of the two operands.   [STRING]
         value(parse_string) = string(first:length)
-------------------------------^
```

**Comment by mcallic2** [comment]
> @mcallic2 looks like an old `value` snuck in from the main merge:
> 
> ```
> mosaic2.F90(399): error #6303: The assignment operation or the binary expression operation is invalid for the data types of the two operands.   [STRING]
>          value(parse_string) = string(first:length)
> -------------------------------^
> ```

should be fixed now

**Review by rem1776 — APPROVED** [review]

**Review by bensonr — APPROVED** [review]

**Pull request merged**